extensions/wikia/WikiFactory/WikiFactory.php

```diff
@@ -1152,59 +1160,42 @@ static public function getLocalEnvURL( $url ) {
 			$address = '/' . $address;
 		}
 
-		// what do we use?
-		//  en.wikiname.wikia.com
-		//  wikiname.wikia.com
-		//  (preview/verify/sandbox).en.wikiname.wikia.com
-		//  (preview/verify/sandbox).wikiname.wikia.com
-		//  en.wikiname.developer.wikia-dev.com
-		//  wikiname.developer.wikia-dev.com
-
-		$servers = [ 'preview.', 'sandbox-s1.', 'verify.' ];
-		foreach ( $servers as $serv ) {
-			if ( strpos( $server, $serv ) === 0 ) {
-				$server = substr( $server, strlen( $serv ) );
-			}
-		}
-
-		$regexp = '/\.([^\.]+)\.wikia-dev\.com$/';
-		if ( preg_match( $regexp, $server, $groups ) === 1 ) {
-			// devbox
-			$devbox = $groups[1];
-			$server = str_replace( '.' . $devbox . '.wikia-dev.com', '', $server );
-		} else {
-			$server = str_replace( '.wikia.com', '', $server );
+		// strip env-specific pre- and suffixes for staging environment
+		$server = mb_ereg_replace( '^(preview|verify|sandbox-[a-z0-9]+)\.', '', $server );
+		if( $wgWikiaEnvironment === WIKIA_ENV_DEV ) {
+			$server = str_replace( $wgDevelEnvironmentName . '.wikia-dev.com', '', $server );
```

Comment:
This needs to be handled differently to be able to transform URL from context of devboxA to devboxB